### ISSUE TITLE
zsh: don't setopt shwordsplit

### ISSUE THREAD

**Issue by marshallpierce** [sdkman/sdkman-cli#210]
In gvm-init.sh, there's this:

```
# force zsh to behave well
if [[ -n $(echo "$SHELL" | grep 'zsh') ]]; then
    setopt shwordsplit
fi
```

It's not polite to setopt shwordsplit. It isn't the zsh default, and setting that broke stuff in my config (and I'm sure other people's, since most people rely on zsh defaults).

Also, [ -n "$ZSH_VERSION" ] is a simpler way to test for zsh.

**Comment by raxod502** [comment]
I recently found this issue because I used SDKMAN to install Gradle and suddenly my (Z) shell behavior broke.

I agree with @marshallpierce: it's *extremely* rude to set shell options and not reset them back to their original values after you're done.

This took quite some tracking down, since having SDKMAN config run at the end of my .zshrc meant I only saw the (wrong) behavior in my running shells, and not when my .zshrc config ran.

**Comment by marc0der** [comment]
@marshallpierce  I'd be keen to have this fixed, feel free to raise a PR.

**Comment by marc0der** [comment]
A big apology for not getting back regarding this earlier.

I've finally had a chance to dig into this a little more and think I have a workable solution. I will check to see if the current shell is ZSH, then use the @marshallpierce approach of set splitting with `SDKMAN_CANDIDATES=( ${(s:,:)SDKMAN_CANDIDATES_CSV} )`, otherwise if bash, we will use the old approach of using a backup of `IFS` to perform splitting.

I'll also introduce new shell flags like we have done for [os support](https://github.com/sdkman/sdkman-cli/blob/master/src/main/bash/sdkman-init.sh#L50), which will make dealing with shell specific issues easier in the future.

**Comment by marc0der** [comment]
@marshallpierce I've pushed this out to our beta channel where it can be tested. Could you confirm that it's working correctly for you now?

**Comment by marshallpierce** [comment]
Yep, it no longer mangles the rest of my shell init! Thanks! :)

**Comment by marc0der** [comment]
Great news! I'll roll it out as 5.5.6 shortly.

Please also let me know if it resolves #531 for you.
